pkg/api/mlflow/service/experiment/service.go

```diff
@@ -231,12 +248,16 @@ func (s Service) SetExperimentTag(ctx context.Context, req *request.SetExperimen
 }
 
 func (s Service) SearchExperiments(
-	ctx context.Context, req *request.SearchExperimentsRequest,
+	ctx context.Context, ns *models.Namespace, req *request.SearchExperimentsRequest,
 ) ([]models.Experiment, int, int, error) {
 	if err := ValidateSearchExperimentsRequest(req); err != nil {
 		return nil, 0, 0, err
 	}
 
+	query := database.DB.Where(
+		"experiments.namespace_id = ?", ns.ID,
+	)
```

Comment:
i like this approach, where we separate the namespace narrowing from the rest of the query